vg/draw/draw_test.go

```diff
@@ -57,32 +58,40 @@ func TestCrop(t *testing.T) {
 	}
 }
 
-func TestSplitVertical(t *testing.T) {
-	var r recorder.Canvas
-	c := NewCanvas(&r, 6, 3)
-	lower, upper := SplitVertical(c, 1)
-	lowerRect := vg.Rectangle{Min: vg.Point{X: 0, Y: 0}, Max: vg.Point{X: 6, Y: 1}}
-	upperRect := vg.Rectangle{Min: vg.Point{X: 0, Y: 1}, Max: vg.Point{X: 6, Y: 3}}
-	if lowerRect != lower.Rectangle {
-		t.Errorf("lower: have %+v, want %+v", lower.Rectangle, lowerRect)
-	}
-	if upperRect != upper.Rectangle {
-		t.Errorf("upper: have %+v, want %+v", upper.Rectangle, upperRect)
+// This example splits a Canvas along a vertical line.
```

Comment:
If you place these each in a separate ..._example_test.go you can reference a package level func rather than using a func var. These don't need to have output comments, just have the code.

So you'll end up with something like, "split_horiz_example_test.go":

```
// Copyright ©2017 The gonum Authors. All rights reserved.
// Use of this source code is governed by a BSD-style
// license that can be found in the LICENSE file.

package draw_test

import (
	"fmt"

	"github.com/gonum/plot/vg"
	"github.com/gonum/plot/vg/draw"
)

// SplitHorizontal returns the left and right portions of c after splitting it
// along a vertical line distance x from the left of c.
func SplitHorizontal(c draw.Canvas, x vg.Length) (left, right draw.Canvas) {
	return draw.Crop(c, 0, c.Min.X-c.Max.X+x, 0, 0), draw.Crop(c, x, 0, 0, 0)
}

func ExampleCrop_splitHorizontal() {
	var c draw.Canvas
	// Split c along a vertical line centered on the canvas.
	left, right := SplitHorizontal(c, c.Size().X/2)
	fmt.Println(left.Rectangle.Size(), right.Rectangle.Size())
}
```

and the equivalent file for split vertical.